front/lib/api/assistant/configuration/table_query.ts

```diff
@@ -77,3 +87,90 @@ export async function fetchTableQueryActionConfigurations({
 
   return actionsByConfigurationId;
 }
+
+export async function createTableDataSourceConfiguration(
+  auth: Authenticator,
+  tableConfigurations: TableDataSourceConfiguration[],
+  tablesQueryConfig: AgentTablesQueryConfiguration,
+  t: Transaction
+) {
+  // Although we have the capability to support multiple workspaces,
+  // currently, we only support one workspace, which is the one the user is in.
+  // This allows us to use the current authenticator to fetch resources.
+  const allWorkspaceIds = [
+    ...new Set(tableConfigurations.map((tc) => tc.workspaceId)),
+  ];
+  const hasUniqueAccessibleWorkspace =
+    allWorkspaceIds.length === 1 &&
+    auth.getNonNullableWorkspace().sId === allWorkspaceIds[0];
+
+  assert(
```

Comment:
Using asserts here as all of this should not happen.

Reply:
We can be much more compact (and I would argue clearer) with:
```
assert(tableConfigurations.every((tc) => tc.workspaceId === auth.getNonNullableWorkspace().sId)
```